api/napi/cover.md

```diff
@@ -0,0 +1,166 @@
+
+```js
+import * as slint from "slint-ui";
+let ui = slint.loadFile(".ui/main.slint");
+let main = new ui.Main();
+main.run();
+```
+
+See [/examples/todo/node](https://github.com/slint-ui/slint/tree/master/examples/todo/node) for a full example.
+
+## API Overview
+
+### Instantiating a component
+
+The exported component is exposed as a type constructor. The type constructor takes as parameter
+an object which allow to initialize the value of public properties or callbacks.
+
+```js
+import * as slint from "slint-ui";
+// In this example, the main.slint file exports a module which
+// has a counter property and a clicked callback
+let ui = slint.loadFile("ui/main.slint");
+let component = new ui.MainWindow({
+    counter: 42,
+    clicked: function() { console.log("hello"); }
+});
+```
+
+### Accessing a property
+
+Properties declared as `out` or `in-out` in `.slint` files are visible as JavaScript on the component instance.
+
+```js
+component.counter = 42;
+console.log(component.counter);
+```
+
+### Callbacks
+
+The callbacks are also exposed as property that has a setHandler function, and that can be called as a function.
+
+```js
+// connect to a callback
+component.clicked.setHandler(function() { console.log("hello"); })
+// emit a callback
+component.clicked();
+```
+
+### Type Mappings
+
+| `.slint` Type | JavaScript Type | Notes |
+| --- | --- | --- |
+| `int` | `Number` | |
+| `float` | `Number` | |
+| `string` | `String` | |
+| `color` | `Color` |  |
+| `brush` | `Brush` |  |
+| `length` | `Number` | |
+| `physical_length` | `Number` | |
+| `duration` | `Number` | The number of milliseconds |
+| `angle` | `Number` | The value in degrees |
+| structure | `Object` | Structures are mapped to JavaScrip objects with structure fields mapped to properties. |
+| array | `Array` or Model Object | |
+
+### Models
+
+For property of array type, they can either be set using an array.
+In that case, getting the property also return an array.
+If the array was set within the .slint file, the array can be obtained
+
+```js
+component.model = [1, 2, 3];
+// component.model.push(4); // does not work, because it operate on a copy
+// but re-assigning works
+component.model = component.model.concat(4);
+```
+
+Another option is to set a model object.  A model object has the following function:
+
+* `rowCount()`: returns the number of element in the model.
+* `rowData(index)`: return the row at the given index
+* `setRowData(index, data)`: called when the model need to be changed. `this.notify.rowDataChanged` must be called if successful.
+
+When such an object is set to a model property, it gets a new `notify` object with the following function
+
+* `rowDataChanged(index)`: notify the view that the row was changed.
+* `rowAdded(index, count)`: notify the view that rows were added.
+* `rowRemoved(index, count)`: notify the view that a row were removed.
+* `reset()`: notify the view that everything may have changed.
+
```

Comment:
I think these belong into the type documentation of `Model<T>` and `ModelPeer<T>`, referencing each other.

```suggestion
```